numba/typeinfer.py

```diff
@@ -819,12 +819,17 @@ def resolve_value_type(self, inst, val):
         Resolve the type of a simple Python value, such as can be
         represented by literals.
         """
-        ty = self.context.resolve_value_type(val)
-        if ty is None:
-            msg = "Unsupported Python value %r" % (val,)
-            raise TypingError(msg, loc=inst.loc)
+        try:
+            ty = self.context.resolve_value_type(val)
+        except ValueError as e:
+            msg = str(e)
+            del e
         else:
-            return ty
+            if ty is not None:
+                return ty
+            msg = "Unsupported Python value %r" % (val,)
+        del val
```

Comment:
Is the delete to avoid `raise` capturing `val` in the frame?  and also for `del e`.  What if it is not deleted?  Shouldn't the reference be dropped as soon as the exception is handled?